lib/rooms/server/scene.js

```diff
@@ -523,6 +525,37 @@ class RoomScene extends RoomLogin
         }
     }
 
+    updateStateSceneData()
+    {
+        let stateSceneData = sc.toJson(this.state.sceneData);
+        this.updatePreloadAssets(stateSceneData);
+        stateSceneData.objectsAnimationsData = this.updateObjectsAnimationData(stateSceneData.objectsAnimationsData);
+        this.state.sceneData = JSON.stringify(stateSceneData);
+    }
+
+    updateObjectsAnimationData(objectsAnimationsData)
+    {
+        for (let key of Object.keys(this.roomData.objectsAnimationsData)) {
```

Comment:
instead of looping all the objects and adding the missing ones we should be adding the objects to the state as soon these are created and removed as soon these are picked up or time-out removed

Reply:
Done!